15.md

```diff
@@ -243,6 +247,110 @@ Create a customized user experience using the `naddr` from [NIP-19](https://gith
+```
+
+NB: items sold as an auction are very similar in structure to fixed-price items, with some important differences worth noting.
+
+* While fixed-price items can be priced in any currency and the conversion to Bitcoin happens on checkout, bids on items sold as an auction happen in sats. Client implementations will probably display conversions to various fiat currencies, but the bidder always commits to an amount to be paid in sats if they will happen to be the top bidder at the end of the auction.
+
+* The `start_date` can be set to a date in the future if the auction is scheduled to start on that date, or can be left `null` if the start date is unknown/hidden. If the start date is `null`, the auction will have to be edited later to set an actual date.
+
+* The auction runs for an initial number of seconds after the `start_date`, specified by `duration`.
+
+### Event `1021`: Bid
+
+```json
+{
+    "content": <int, amount of sats>,
+    "tags": [["e", <event ID of the auction to bid on>]],
+}
+```
+
+Bids are simply events of kind `1021` with a `content` field specifying the amount, in sats. Bids must reference an auction.
+
+NB: auctions can be edited as many times as desired (they are "parameterized replaceable events") by the author - even after the start_date, but they cannot be edited after they have received the first bid! This is enforced by the fact that bids reference the event ID of the auction (rather than the product UUID), which changes with every new version of the auctioned product. So a bid is always attached to one "version". Editing the auction after a bid would result in the new product losing the bid!
+
+### Event `1022`: Bid confirmation
+
+```json
+{
+    "content": <json>
+    "tags": [["e" <event ID of the bid being confirmed>], ["e", <event ID of the auction>]],
+}
+```
+
+Bids should be confirmed by the merchant before being considered as valid by other clients. So clients should subscribe to *bid confirmation* events (kind `1022`) for every auction that they follow, in addition to the actual bids and should check that the pubkey of the bid confirmation matches the pubkey of the merchant (in addition to checking the signature).
+
+The `content` field is a JSON which includes *at least* a `status`, which is one of `accepted`, `rejected`, `pending`, or `winner`. `winner` is how the *winning bid* is replied to after the auction ends and the winning bid is picked by the merchant.
+
+The reasons for which a bid can be marked as `rejected` or `pending` are up to the merchant's implementation and configuration - they could be anything from basic validation errors (amount too low) to the bidder being blacklisted or to the bidder lacking sufficient *trust*, which could lead to the bid being marked as `pending` until sufficient verification is performed. The difference between the two is that `pending` bids *might* get approved after additional steps are taken by the bidder, whereas `rejected` bids can not be later approved.
```

Comment:
"Merchants verify bidders" by (possibly) asking for a deposit is - sort of - what we do, amongst others! We call it "skin in the game", which is a Nostr (NIP-58) badge you can purchase. You can totally win an auction without revealing your real identity.

All in all, this is probably the most complex issue when it comes to Nostr auctions and I think we are too early to set it in stone at the protocol level. We might to it later, as we learn more and gather more feedback.

But mentioning a "deposit" as part of the NIP is - I feel - encouraging implementations to be money handlers, which is a dangerous path to go. Unless you say that the deposits stay in the merchant's hands. But then - we need another protocol to handle the refunds.

Reply:
We are believers in "reputation by participation". Real world auctions don't (really) allow this, but Nostr does. You can be a keypair with nothing else known about you, just a history of having successfully bought stuff in the past. And the history is known by the marketplace.

The issue really is newcomers. Fresh keypairs need *some* way to acquire trust. Known keypairs are fine, because the more reputation you have, the more likely you want to hold on to that and not lose it by screwing up and have your whole reputation slashed!

That's why the decision is not really done at the protocol level, but rather at the implementation level. How can you define what is enough "skin in the game" that makes you unlikely to want to be a troll?